dynawo/sources/Modeler/DataInterface/DYNCriteria.cpp

```diff
@@ -94,13 +144,62 @@ BusCriteria::addBus(const boost::shared_ptr<BusInterface>& bus) {
   buses_.push_back(bus);
 }
 
+BusCriteria::BusFailingCriteria::BusFailingCriteria(Bound bound,
+                                                    std::string busId,
+                                                    double v,
+                                                    double vPu,
+                                                    double uBound,
+                                                    double uBoundPu,
+                                                    const std::string criteriaId) :
+  FailingCriteria(bound, busId, criteriaId),
+  v_(v),
+  vPu_(vPu),
+  uBound_(uBound),
+  uBoundPu_(uBoundPu) {}
+
+void
+BusCriteria::BusFailingCriteria::printOneFailingCriteriaIntoLog() const {
```

Comment:
Do we really need to define 2 differente methods here?

Reply:
I think so... the attributes of BusFailingCriteria and LoadFailingCriteria aren't the same. I don't think we can use the same method in both cases.

Reply:
I mean 2 methods, not 2 classes, cannot we merge the dump of timeline and log into a single method printOneFailingCriteria (bool dumpTImeline) ?